### ISSUE TITLE
Where are all the READMEs?

### ISSUE THREAD

**Issue by billiegoose** [zeke/all-the-packages#3]
I'm running this piece of code to try and count what percent of packages in the registry have README files.

```
let haves = 0
let total = 0

registry.on('package', (pkg) => {
  total++
  if (pkg.readme) haves++
  if (total % 1000 === 0) console.log(`${haves} / ${total}    ${(100 * haves / total).toFixed(1)}%`)
})
registry.on('end', () => {
  console.log(`${haves} / ${total}`)
  console.log('percent =', (100 * haves / total).toFixed(2))
  process.stderr.write('DONE\n')
  process.exit(0)
})
```

Now if I use

```
const registry = require('all-the-packages')
```

I get one answer: 25394 / 412075 or about 6.2%

But if I use 

```
const registry = require('package-stream')()
```

I get a *completely* different answer. Right now I'm only about 1/4 of the way through (100,000 packages) but it's at 95666 READMEs or 95%.

What's going on?

**Comment by zeke** [comment]
I suspect this is because readmes usually live  deeper down in the registry metadata: `pkg.dist-tags.latest`

`package-stream` turns every result into a `nice-package` instance. Nice packages have their `latest` version metadata copied up to the top-level object. This also includes things like `dependencies`

`all-the-packges` is not using `nice-package`, so no extra goodies in the top level of the object.

Does that make sense?

**Comment by billiegoose** [comment]
Ah! That would explain it!

(an aside) you might appreciate this. I really just made this as a sanity check - I wanted to see if there were any weird anomalies in the distribution (There weren't). I counted the # of lines of every readme, sorted the list from small to large, and plotted the result. I had to plot the log of the length though because there is such variation (from 1 line long to thousands of lines long):
![plots](https://cloud.githubusercontent.com/assets/587740/24268096/e4f05e64-0fe2-11e7-8c58-5e56671ac7bf.PNG)

Since the graph's badly labeled I'll explain it some more. The left graph simply says that 86% of the packages had a README. For the right graph, the Y axis is the log of the # of lines long a README file is and the X axis I guess would be the percentile. I only calculated it at 10 points but connected them with a solid line, because I wasn't thinking, which is probably why there's a sharp bend at 89%. The reason the labels are 9%, 19%, etc instead of 10%, 20%, etc is because I made an off-by-one-error. The median README length would be at the 50% on the X axis, which is somewhere between 10 and 100 lines long. The 1 on the Y axis corresponds to 10 lines long, and the plot crosses that at around 20% so roughly 20% or READMEs are less than 10 lines long. By the same reasoning, it looks READMEs over 1000 lines long make up the 5% or so.

As time permits, I'll make pie charts showing what percent make up of READMEs are Markdown vs HTML, English vs Chinese, a word cloud to show the most popular `<h2>` headings etc.
